scijava/scijava-ops-api/templates/main/java/org/scijava/ops/api/OpBuilder.vm

```diff
@@ -47,13 +47,52 @@ import org.scijava.types.Types;
 
 /**
  * Convenience class for looking up and/or executing ops using a builder
- * pattern.
+ * pattern. The general order of specification:
+ * <ol>
+ *   <li>The op name</li>
+ *   <li>The number of input(s)</li>
+ *   <li>The type or value(s) of input(s)</li>
+ *   <li>One of:
+ *     <ul>
+ *       <li>The type or value of the output</li>
+ *       <li>Which input should be modified in-place</li>
+ *     </ul>
+ *   </li>
+ * </ol>
+ * The first two steps are required, at a minimum. The choices you make will
+ * determine the <i>type</i> of Op that is matched:
+ * <ul>
+ *   <li>No inputs &rarr; <code>Producer</code> or <code>Computer</code></li>
+ *   <li>Inputs with an output <i>value</i> &rarr; <code>Computer</code></li>
+ *   <li>Inputs with an output <i>type</i> &rarr; <code>Computer</code> or <code>Function</code></li>
+ *   <li>Inputs with no output &rarr; <code>Function</code> or <code>Inplace</code></li>
+ * </ul>
+ * <br/>
  * <p>
- * TODO: Examples
+ * Note that the intermediate builder steps use the following acronyms:
```

Comment:
OK, so this part I would keep here on the `OpBuilder` javadoc, and not move anywhere more visible. The IV/OV et al acronyms were always intended to be as hidden as possible. Obviously they are public classes, and the IDE will report the return types of things, but that's exactly why I made them as short as possible: because I didn't want people worrying about these types—instead the goal is to focus them on the sequence of chained method calls, and what methods become available to use at each step. The return types are an implementation detail. All of that said, I think the explanation here is nice for the people who are curious.